core/domain/services/admin/entities/DefaultPrices.php

```diff
@@ -55,66 +65,114 @@ public function __construct(EEM_Price $price_model, EEM_Price_Type $price_type_m
      * @throws ReflectionException
      * @since $VID:$
      */
-    public function create(EE_Base_Class $entity)
+    public function create(EE_Base_Class $entity): array
     {
         if (! $entity instanceof EE_Ticket) {
             throw new InvalidEntityException($entity, 'EE_Ticket');
         }
-        $taxes = [];
-        $new_prices = [];
-        $is_free = true;
-        $has_base_price = false;
-        $default_prices = $this->price_model->get_all_default_prices(false, true);
-        if (is_array($default_prices)) {
-            foreach ($default_prices as $default_price) {
-                if (! $default_price instanceof EE_Price) {
-                    throw new InvalidEntityException($default_price, 'EE_Price');
-                }
-                // grab any taxes but don't do anything just yet
-                if ($default_price->is_tax()) {
-                    $taxes[] = $default_price;
-                    continue;
-                }
-                // duplicate the default price so that it does not get mutated
-                /** @var EE_Price $default_price_clone */
-                $default_price_clone = clone $default_price;
-                if ((
+        $is_free                   = true;
+        $has_base_price            = false;
+        $remove_existing_relations = true;
+        // first, let's get all of the default taxes for the site
+        $this->taxes = $this->price_model->getAllDefaultTaxes();
+        // then separate taxes from the other prices for the existing default ticket prices
+        $default_prices = $this->separateTaxesFromOtherPrices($entity->prices());
+        // but if that's empty, then let's get the default global prices
+        if (empty($default_prices)) {
+            $default_global_prices     = $this->price_model->get_all_default_prices();
+            $default_prices            = $this->separateTaxesFromOtherPrices($default_global_prices);
+            $remove_existing_relations = false;
+        }
+        // then clone and apply all of the default prices
+        [$has_base_price, $is_free] = $this->cloneDefaultPrices(
+            $entity,
+            $default_prices,
+            $remove_existing_relations,
+            $has_base_price,
+            $is_free
+        );
+        if (! $has_base_price) {
+            $new_base_price                            = $this->createNewBasePrice($entity);
+            $this->new_prices[ $new_base_price->ID() ] = $new_base_price;
+        }
+        $this->applyTaxes($entity, $is_free);
```

Comment:
Does this modify `$entity` inn place? 🤔

Reply:
yes but the entity (a ticket in this case) is a clone which was created for the sole purpose of mutating